src/t_stream.c

```diff
@@ -1862,8 +1888,9 @@ void xaddCommand(client *c) {
 
     /* Let's rewrite the ID argument with the one actually generated for
      * AOF/replication propagation. */
+    int idpos = field_pos-1;  /* The ID is always one argument before the first field */
     robj *idarg = createObjectFromStreamID(&id);
-    rewriteClientCommandArgument(c,i,idarg);
+    rewriteClientCommandArgument(c,idpos,idarg);
```

Comment:
Oops, I didn't realize that ID is also after what's handled by streamParseAddOrTrimArguments()... So maybe I change my mind and instead I think streamParseAddOrTrimArguments() should just `return i`. The point of the previous comment was to return the position after the args handled by the function but I didn't realize that `return i` already did that, so I'm sorry for this review mistake!

If you agree, the code in xaddCommand() would be something like the below (like in the first version, just renaming i to idpos). What do you think?

```c
    int idpos = streamParseAddOrTrimArguments(c, &parsed_args, 1);
    if (idpos < 0) return; /* streamParseAddOrTrimArguments already replied. */
    int field_pos = idpos+1;
```